### PR TITLE
Integrate some functions in KVStore and rename them to what they actually do

### PR DESCRIPTION
**Pull request by CalvinNeo**
### What problem does this PR solve?

Issue Number: ref #7256

Problem Summary:

Some methods are delegators of `writeBlockByRegion` which transforms row-format data into columnar-format data, and write them to cache of DeltaMergeStore.
There methods don't always do a flush. **Actually, they ALWAYS NOT DO a flush**, the only exception is when they are called from `tryFlushRegions`, and that is purely a scene of debugger.

1. flushRegion
2. tryPersist this is some un-locked version of persistRegion, we change its name to tryPersistRegion, and make it coworks with persistRegion.
4. tryFlushRegion
5. tryFlushRegions

So these names are quite misleading, I think we have to change them to what they actually do. 

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Debug/MockRaftStoreProxy.cpp [hunk 1]
```diff
@@ -547,7 +547,7 @@ void MockRaftStoreProxy::doApply(
         return;
 
     auto old_applied = kvs.getRegion(region_id)->appliedIndex();
-    auto old_applied_term = kvs.getRegion(region_id)->appliedIndexTerm();
```
Comment (CalvinNeo) [inline dbms/src/Debug/MockRaftStoreProxy.cpp:550]: typo

File: dbms/src/Debug/dbgFuncRegion.cpp [hunk 2]
```diff
@@ -101,7 +101,7 @@ void dbgFuncPutRegion(Context & context, const ASTs & args, DBGInvoker::Printer
 void dbgFuncTryFlush(Context & context, const ASTs &, DBGInvoker::Printer output)
 {
     TMTContext & tmt = context.getTMTContext();
-    tmt.getRegionTable().tryFlushRegions();
```
Comment (lidezhu) [inline dbms/src/Debug/dbgFuncRegion.cpp:104]: I think function name should describe what job it intend to do not how to do the job. So I prefer the previous one.
And this function is actually deprecated, so I think there is no need to change its name.

Comment (CalvinNeo) [reply dbms/src/Debug/dbgFuncRegion.cpp:104]: I don't know what job dbgFuncTryFlush intends to do. However, if dbgFuncTryFlush is deprecated, we can remove dbgFuncTryFlush and tryFlushRegions

**Review by lidezhu — APPROVED** [review]
LGTM

File: dbms/src/Storages/Transaction/RegionTable.h [hunk 3]
```diff
@@ -211,7 +216,10 @@ class RegionTable : private boost::noncopyable
     InternalRegion & insertRegion(Table & table, const Region & region);
     InternalRegion & doGetInternalRegion(KeyspaceTableID ks_tb_id, RegionID region_id);
 
-    RegionDataReadInfoList flushRegion(const RegionPtrWithBlock & region, bool try_persist) const;
+    // Try write the committed kvs into cache of columnar DeltaMergeStore.
+    // Flush the cache if try_persist is set to true.
+    // The original name for this method is flushRegion.
```
Comment (JaySon-Huang) [inline dbms/src/Storages/Transaction/RegionTable.h:221]:
```suggestion
```

File: dbms/src/Storages/Transaction/RegionTable.h [hunk 4]
```diff
@@ -159,9 +159,15 @@ class RegionTable : private boost::noncopyable
...
 
+    // Find all regions with data, call writeBlockByRegionAndFlush with try_persist = true.
+    // This function is only for debug.
     bool tryFlushRegions();
-    RegionDataReadInfoList tryFlushRegion(RegionID region_id, bool try_persist = false);
-    RegionDataReadInfoList tryFlushRegion(const RegionPtrWithBlock & region, bool try_persist);
+
+    // Protects writeBlockByRegionAndFlush and ensures it's executed by only one thread at the smae time.
+    // Only one thread can do this at the same time.
+    // The original name for this function is tryFlushRegion.
```
Comment (JaySon-Huang) [inline dbms/src/Storages/Transaction/RegionTable.h:168]:
```suggestion
```

File: dbms/src/Storages/Transaction/KVStore.cpp [hunk 5]
```diff
@@ -333,12 +330,27 @@ void KVStore::setRegionCompactLogConfig(UInt64 sec, UInt64 rows, UInt64 bytes)
...
+        region_persister->persist(region);
+    }
+    if (region_task_lock.has_value())
+    {
+        LOG_DEBUG(log, "Persist {} done", region.toString(false));
+    }
+    else
+    {
+        LOG_INFO(log, "After persisted {}, cache {} bytes", region.toString(false), region.dataSize());
+    }
```
Comment (JaySon-Huang) [inline dbms/src/Storages/Transaction/KVStore.cpp:353]: the statements inside this if branch can be merged into the previous if branch?

File: dbms/src/Storages/Transaction/RegionTable.h (hunk 3 above)
Comment (CalvinNeo) [reply dbms/src/Storages/Transaction/RegionTable.h:221]: Why?

File: dbms/src/Storages/Transaction/KVStore.cpp (hunk 5 above)
Comment (CalvinNeo) [reply dbms/src/Storages/Transaction/KVStore.cpp]: fixed

File: dbms/src/Storages/Transaction/RegionTable.h (hunk 3 above)
Comment (JaySon-Huang) [reply dbms/src/Storages/Transaction/RegionTable.h:221]: I think we don't need to keep this message... it doesn't help

**Review by JaySon-Huang — APPROVED** [review]
LGTM

**Pull request merged**